lib/compilers/dotnet.ts

```diff
@@ -30,28 +30,28 @@ import fs from 'fs-extra';
 import { BaseCompiler } from '../base-compiler';
 
 class DotNetCompiler extends BaseCompiler {
-    private rID: string;
-    private targetFramework: string;
-    private buildConfig: string;
-    private nugetPackagesPath: string;
-    private clrBuildDir: string;
-    private additionalSources: string;
-    private langVersion: string;
+    private coreRoot: string;
+    private crossgen2bin: string;
+    private testAppSrc: string;
+    private testAppName: string;
+    private testAppOutputDir: string;
 
     constructor(compilerInfo, env) {
         super(compilerInfo, env);
 
-        this.rID = this.compilerProps(`compiler.${this.compiler.id}.runtimeId`);
-        this.targetFramework = this.compilerProps(`compiler.${this.compiler.id}.targetFramework`);
-        this.buildConfig = this.compilerProps(`compiler.${this.compiler.id}.buildConfig`);
-        this.nugetPackagesPath = this.compilerProps(`compiler.${this.compiler.id}.nugetPackages`);
-        this.clrBuildDir = this.compilerProps(`compiler.${this.compiler.id}.clrDir`);
-        this.additionalSources = this.compilerProps(`compiler.${this.compiler.id}.additionalSources`);
-        this.langVersion = this.compilerProps(`compiler.${this.compiler.id}.langVersion`);
+        this.coreRoot = this.compilerProps(`compiler.${this.compiler.id}.coreRoot`);
+        this.testAppSrc = this.compilerProps(`compiler.${this.compiler.id}.testAppSrc`);
+        this.testAppName = path.basename(this.testAppSrc);
+        this.crossgen2bin = path.join(this.coreRoot, 'crossgen2', 'crossgen2.dll');
+        this.testAppOutputDir = 'out';
     }
 
     get compilerOptions() {
-        return ['publish', '-c', this.buildConfig, '--self-contained', '--runtime', this.rID, '-v', 'q', '--nologo'];
+        return ['build', this.testAppName + '.csproj', '-c', 'Release', '-o', this.testAppOutputDir, 
+        // Properties:
+        '/p:AllowUnsafeBlocks=true',
+        // Speed up compilation:
+        '--no-restore', '--no-dependencies', '--nologo', '-v', 'q', '--nologo'];
```

Comment:
Do we need two `--nologo` options?

Reply:
I don't think we need any :D the parameters don't actually work, it's the env setting that did the trick for me

Reply:
Removed 🙂